src/reducers/view.js

```diff
@@ -97,26 +135,16 @@ const lastFetch = (state = null, action) => {
 
 export default combineReducers({
   displayedFolder,
+  fileCount,
   files,
   fetchStatus,
   lastFetch
 })
 
-const sortFiles = files => files.sort((a, b) => a.name.localeCompare(b.name))
-
-const getSortedFiles = allFiles => {
-  const folders = allFiles.filter(f => f.type === 'directory' && f.id !== TRASH_DIR_ID && f.path !== APPS_DIR_PATH && f.path !== KONNECTORS_DIR_PATH)
-  const files = allFiles.filter(f => f.type !== 'directory')
-  return sortFiles(folders).concat(sortFiles(files))
-}
-
-export const getVisibleFiles = ({ view }) => {
-  const { files } = view
-  return getSortedFiles(files)
-}
+export const getVisibleFiles = ({ view }) => view.files
 
 export const getFileById = ({ view }, id) => {
-  const file = view.files.find(f => f.id === id)
+  const file = view.files.find(f => f && f.id && f.id === id)
```

Comment:
Are you sure is nessacerry `&& f.id &&`:

    const file = view.files.find(f => f && f.id === id)